pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -407,6 +413,11 @@ func (a *Actuator) getHost(ctx context.Context, machine *machinev1beta1.Machine)
 		return nil, nil
 	} else if err != nil {
 		return nil, err
+	} else if uid != nil {
```

Comment:
If the UID is nil, one of two things has happened.
(a) we haven't reached the Provisioned phase yet, and we're still calling Create() and relying on the annotation (not the ProviderID) to tell us which Host we've selected; or
(b) the ProviderID was set by an older version of CAPBM that doesn't include the UID (we only started setting the ProviderID in 4.6, but this change doesn't look likely to get in before 4.7 so this will happen).

We don't add the UID to the annotation because there's not much point; there can't be a Node yet and the goal is to differentiate between Nodes created from different Hosts.
It would be nice to update old ProviderIDs (and I wrote and then deleted code to do that), but it transpires that we can't because it's immutable - there's a webhook that stops us from changing it on the Node (specifically, it can only change from an empty string to a not empty string).

Reply:
These are the sorts of assumptions I like to spell out in comments. I'm not sure if this is the right spot in the code to do that, but it's the point at which I had the question in the review, so maybe?

Reply:
I'll add a comment in `ensureMachineProviderID()`